app/src/queries/useEstimatedTimestamp.ts

```diff
@@ -0,0 +1,25 @@
+import { useAnchorWebapp } from '@anchor-protocol/app-provider';
+import { useLastSyncedBlock } from './useLastSyncedBlock';
+import { millisecondsInHour, getDaysInYear } from 'date-fns';
+
+const hoursInDay = 24;
+
+export const useEstimatedTimestamp = (
+  blockHeight: number,
+): number | undefined => {
+  const {
+    constants: { blocksPerYear },
+  } = useAnchorWebapp();
+
+  const { data: lastSyncedBlock } = useLastSyncedBlock();
+  if (!lastSyncedBlock) return;
+
+  const daysInYear = getDaysInYear(new Date());
```

Comment:
It won't be super corrected for a block that is not in the same year, but it's not significant in the current use case.

Reply:
I'm not a fan omitting the braces, at some point I will add that to the linting rules so we may aswell add it now, so

`if (!lastSyncedBlock) return;` should be

```
if (!lastSyncedBlock) {
  return;
}
```

Also I avoid using the negating operator where possible as its too easy to miss when reading code, so this is way more explicit

```
if (lastSyncedBlock === undefined) {
  return;
}
```

Reply:
✅ https://github.com/Anchor-Protocol/anchor-web-app/pull/309/commits/a39b86bccd46575eee5ea009a7eaae3f392f7fba